pd/src/app.rs

```diff
@@ -87,6 +87,11 @@ impl App {
         let note_commitment_tree = state.note_commitment_tree().await?;
         let genesis_config = state.genesis_configuration().await?;
         let recent_anchors = state.recent_anchors(NUM_RECENT_ANCHORS).await?;
+        // TODO: This is the only place `state.validators()` is called -- updates to
+        // the validator list won't be available! I think it needs to be called again in `commit`
+        // prior to updating validator rates. _Or_ maybe not depending on the database interaction design.
+        // But we need to be careful to ensure the `App.validators` doesn't come out of sync with what's
+        // in the database.
```

Comment:
this is a good concern, and (non-blocking on this PR) it's probably worth removing the `validators` field from the `App` for now, to prevent inconsistency.  in the event that we need to access the validator data more frequently, we can add it back and ensure consistency then.

Reply:
👍